File: src/main/kotlin/net/yttravelers/spotshare/entities/RouteDetailHotel.kt
package net.yttravelers.spotshare.entities

import com.fasterxml.jackson.annotation.JsonIgnore
import java.time.LocalDateTime
import javax.persistence.Table
import javax.persistence.*

@Entity
@Table(name = "routeDetailHotel")
class RouteDetailHotel {
    @Id
    var routeDetailId: Int? = null

    @JsonIgnore
    @OneToOne
    @JoinColumn(name = "routeDetailId")
    @MapsId
    lateinit var routeDetail: RouteDetail

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "hotelId")
    lateinit var hotel: Hotel

    var hotelRate: Int? = null

    var hotelMinutes: Int? = null

    //TODO: 区分の設計
//    var hotelKindDiv:
    //必要？
//    var hotelKindDivName:

    var hotelCheckInTime: LocalDateTime? = null

    var hotelCheckOutTime: LocalDateTime? = null

    //TODO: 区分の設計
//    var hotelBreakfastYesNoDiv:

    //TODO: 区分の設計
//    var hotelDinnerYesNoDiv:

    var hotelCost: Int? = null
}

File: src/main/angular/src/app/model/route-detail-hotel.ts
import { IHotel } from "./hotel";

/**
 * ルート詳細ホテル
 */
export interface IRouteDetailHotel {
  /** ルート詳細ID */
  routeDetailId?: number;
  /** ホテルID */
  hotelId?: number;
  /** 行きたい度 */
  hotelRate?: number;
  /** 滞在時間 */
  hotelMinutes?: number;
  /** チェックイン時間（時） */
  hotelCheckInTimeHours?: string;
  /** チェックイン時間（分） */
  hotelCheckInTimeMinutes?: string;
  /** チェックアウト時間（時） */
  hotelCheckOutTimeHours?: string;
  /** チェックアウト時間（分） */
  hotelCheckOutTimeMinutes?: string;
  /** 朝食有無区分 */
  hotelBreakfastYesNoDiv?: number;
  /** 朝食有無区分名称 */
  hotelBreakfastYesNoDivName?: string;
  /** 夕食有無区分 */
  hotelDinnerYesNoDiv?: number;
  /** 夕食有無区分名称 */
  hotelDinnerYesNoDivName?: string;
  /** 宿泊費用 */
  hotelCost?: number;
  /** ホテル */
  hotel?: IHotel;
}